tests/test_scripts.py

```diff
@@ -54,6 +55,24 @@ def test_load_into_counting():
     assert os.path.exists(outfile)
 
 
+def test_load_into_counting_nonwritable():
+    script = scriptpath('load-into-counting.py')
+    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
+
+    outfile = utils.get_temp_filename('test-nonwritable')
+    with open(outfile, 'w') as fout:
+        fout.write("This file is non-writable (after this)")
+
+    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
+    infile = utils.get_test_data('test-abund-read-2.fa')
+
+    args.extend([outfile, infile])
+
+    (status, out, err) = utils.runscript(script, args, fail_ok=True)
+    assert 'does not have write permission; exiting' in err, err
```

Comment:
Stop guessing and look :-) Does it clean up or not when you run it?

On Tue, Apr 7, 2015, 17:32 Jake Fenton <notifications@github.com> wrote:

> In tests/test_scripts.py
> <https://github.com/ged-lab/khmer/pull/892#discussion_r27922849>:
>
> > @@ -54,6 +55,24 @@ def test_load_into_counting():
> >      assert os.path.exists(outfile)
> >
> >
> > +def test_load_into_counting_nonwritable():
> > +    script = scriptpath('load-into-counting.py')
> > +    args = ['-x', '1e3', '-N', '2', '-k', '20', '-t']
> > +
> > +    outfile = utils.get_temp_filename('test-nonwritable')
> > +    with open(outfile, 'w') as fout:
> > +        fout.write("This file is non-writable (after this)")
> > +
> > +    os.chmod(outfile, stat.S_IWOTH | stat.S_IRUSR)
> > +    infile = utils.get_test_data('test-abund-read-2.fa')
> > +
> > +    args.extend([outfile, infile])
> > +
> > +    (status, out, err) = utils.runscript(script, args, fail_ok=True)
> > +    assert 'does not have write permission; exiting' in err, err
>
> Like all tempfiles it's in the utils' clean list. I'd normally say "since
> it isn't spewing errors everywhere I assume it was successfully removed"
> but now that you're asking I'm suspicious--I could restore write
> permissions at the end of the test and then allow the normal autoclean to
> take place.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ged-lab/khmer/pull/892/files#r27922849>.
>